boa/src/syntax/parser/mod.rs

```diff
@@ -876,41 +1015,1330 @@ impl Parser {
+            {
+                Ok(succ) => succ,
+                Err(ParseError::NormalEOF) => false,
+                Err(e) => return Err(e),
+            } {}
+        }
+
+        if is_block_statement {
+            Ok(Node::Block(items))
+        } else {
+            Ok(Node::StatementList(items))
+        }
+    }
+
+    /// <https://tc39.es/ecma262/#prod-StatementListItem>
+    fn read_statement_list_item(&mut self) -> Result<Node, ParseError> {
+        if let Ok(tok) = self.peek_skip_lineterminator() {
+            match tok.kind {
+                TokenKind::Keyword(Keyword::Function) => self.read_declaration(),
+                TokenKind::Keyword(Keyword::Const) => self.read_declaration(),
+                TokenKind::Keyword(Keyword::Let) => self.read_declaration(),
+                _ => self.read_statement(),
+            }
+        } else {
+            Err(ParseError::NormalEOF)
+        }
+    }
+
+    /// <https://tc39.es/ecma262/#prod-Declaration>
+    fn read_declaration(&mut self) -> Result<Node, ParseError> {
+        let tok = self.next_skip_lineterminator()?;
+        match tok.kind {
+            TokenKind::Keyword(Keyword::Function) => self.read_function_declaration(),
+            TokenKind::Keyword(Keyword::Const) => self.read_lexical_declaration(true),
+            TokenKind::Keyword(Keyword::Let) => self.read_lexical_declaration(false),
+            _ => unreachable!(),
+        }
+    }
+
+    /// <https://tc39.es/ecma262/#prod-LexicalDeclaration>
+    fn read_lexical_declaration(&mut self, is_const: bool) -> Result<Node, ParseError> {
+        let mut list = vec![];
+
+        loop {
+            let token = self.next_skip_lineterminator()?;
+            let name = match token.clone().kind {
+                TokenKind::Identifier(name) => name,
+                _ => {
+                    return Err(ParseError::Expected(
+                        vec![TokenKind::Identifier(String::from("Identifier"))],
+                        token.clone(),
+                        "Expect identifier.",
+                    ));
+                }
+            };
+
+            if self.next_if_skip_lineterminator(TokenKind::Punctuator(Punctuator::Assign))? {
+                let init = Some(self.read_initializer()?);
+                let decl = if is_const {
+                    Node::ConstDecl(vec![(name, init.unwrap())])
+                } else {
+                    Node::LetDecl(vec![(name, init)])
+                };
+
+                list.push(decl);
+            } else {
+                if is_const {
+                    return Err(ParseError::Expected(
+                        vec![TokenKind::Identifier(String::from("Expression"))],
+                        token,
+                        "Expected Expression for Const!",
+                    ));
+                } else {
+                    list.push(Node::LetDecl(vec![(name, None)]));
+                }
+            }
```

Comment:
This should fix the clippy error:
```suggestion
            } else if is_const {
                return Err(ParseError::Expected(
                    vec![TokenKind::Identifier(String::from("Expression"))],
                    token,
                    "Expected Expression for Const!",
                ));
            } else {
                list.push(Node::LetDecl(vec![(name, None)]));
            }
```